src/ess.jl

```diff
@@ -296,15 +300,17 @@ function ess_rhat(
             # generate a monotone sequence
             pₜ = min(Δ, pₜ)
 
-            # update sum
-            sum_pₜ += pₜ
-
             # update indices
             k += 2
         end
 
+        # improved truncation reduces variance for antithetic chains
+        τ = 2 * sum_pₜ + max(0, ρ_even) - 1
+
+        # enforce ESS is in [0, ntotal*log10(ntotal)]
```

Comment:
Wouldn't it be more natural to enforce this on ESS below instead of tau?

Reply:
It requires an additional conditional. This approach sets both negative ESS and large ESS to the maximum value of `ntotal*log10(ntotal)`. Otherwise we would use
```julia
ess_max = LogExpFunctions.xlogx(oftype(one(T), ntotal))
ess[i] = τ < 0 ? ess_max : min(ntotal / τ, ess_max) 
```

Reply:
Ah I didn't realize that negative values are not set to 0 but also the maximum value. It could be written also as
```julia
ess_i = ntotal / max(τ, zero(τ))
ess_max = ntotal * log10(oftype(ess_i, ntotal))
ess[i] = min(ess_i, ess_max)
```
(ideally with a precomputed `ess_max` of the correct type) Still requires an additional conditional but arguably looks nicer :smile:

Reply:
Yeah, I agree this is clearer.